### PR TITLE
Changelog 24.7.0

### PR DESCRIPTION
**Pull request by beeankha**
## Description

✂️ snip snip ✂️ the making of a new release.

Xref #5381

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,133 @@
 [//]: # (current developments)
 
+## 24.7.0 (2024-07-18)
+
+### Enhancements
+
+* Skip generating `repodata.json.bz2` for local index; generate `repodata.json`
+  only; require `conda-package-handling >=2.2.0` matching conda. (#5231)
+* Add new include/exclude sections for glob expressions in multi-output `outputs/files`. (#4196 via #5216)
+* Increased performance by using pickle instead of deepcopy. (#5281)
```
Comment (kenodegard) [inline CHANGELOG.md:10]:
```suggestion
* Increase performance by using `pickle` instead of `copy.deepcopy`. (#5281)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,133 @@
...
+* Report fully rendered recipe to stdout before the build process starts. (#3798 via #5344)
+* Validate `run_constrained` dependencies to prevent faulty specs reaching final repodata. (#5047 via #5359)
+* PIP_* env variables are set when building outputs in multi-output recipes. (#3993 via #5368)
```
Comment (kenodegard) [inline CHANGELOG.md:13]:
```suggestion
* `PIP_*` env variables are set when building outputs in multi-output recipes. (#3993 via #5368)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,133 @@
...
+* Reduced performance overheads of logging. (#5384)
```
Comment (kenodegard) [inline CHANGELOG.md:14]:
```suggestion
* Reduce performance overhead of logging. (#5384)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,133 @@
...
+  * `conda_build.metadata.toposort` (use `conda_build.metadata.toposort_outputs` instead). (#5406)
+  * `conda_build.utils._convert_lists_to_sets` (use `frozendict.deepfreeze` instead). (#5333)
+  * `conda_build.utils.HashableDict` (use `frozendict.deepfreeze` instead). (#5333)
+  * `conda_build.utils.represent_hashabledict` (use `frozendict.deepfreeze` instead). (#5333)
+  * `conda_build.utils.rm_rf(config)`. (#5333)
+  * `conda_build.variants.get_vars(loop_only)`. (#5333)
+
+### Other
+
+* Long `conda info` error messages for common exceptions. (#5264)
```
Comment (kenodegard) [inline CHANGELOG.md:109]: Long? Log?

Comment (beeankha) [reply CHANGELOG.md:109]:
```suggestion
* Log `conda info` error messages for common exceptions. (#5264)
```
🪵

Comment (kenodegard) [reply CHANGELOG.md:109]:
```suggestion
* Do not generate conda error reports for common exceptions. (#5264)
```

Comment (beeankha) [reply CHANGELOG.md:109]: Oops, good catch! I cut + pasted the issue title, which is what got fixed (the error reports were too long)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,133 @@
...
+* Add `CondaBuildUserError` exception to replace `sys.exit` calls. (#5353)
+* Updated the CI to download the MacOSX 10.15 SDK.
```
Comment (kenodegard) [inline CHANGELOG.md:111]:
```suggestion
* Updated the CI to download the MacOSX 10.15 SDK. (#5387)
```

**Review by kenodegard — APPROVED** [review]

**Pull request merged**